tests/functional/tests/smoke.py

```diff
@@ -122,6 +122,21 @@ def cou(self, cmd: list[str]) -> CompletedProcess:
         """
         return run([self.exc_path] + cmd, capture_output=True, text=True)
 
+    def test_help(self) -> None:
+        """Test that help command is working."""
+        cmds = ["", "plan", "run"]
+        help_options = ["-h", "--help"]
+        for cmd in cmds:
+            for help in help_options:
```

Comment:
Exactly for this I prefer to use pytest. Now it's too late, but I would be better to use fixtures for installing snap and cou wrapper.

Reply:
Do you mean the for-loop?

Reply:
I was going to suggest that, because I saw it in hw-observer PR. But when I was looking on it at pypi.org I found out that this package is not maintained often (last change is from 10 months ago and previous one is 3 years old). That's why I do not think that it's good to be used, I spoke about it with @dashmage.

Reply:
It's not like pytest fixture, but using [SubTest](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests) is a better approach because it will show which sub test failed.

I'll change to something like:
```python
    def test_help(self) -> None:
        """Test that help command is working."""
        help_options = itertools.product(["", "plan", "run"], ["-h", "--help"])
        for cmd, help in help_options:
            help_cmd = [cmd, help] if cmd else [help]
            with self.subTest(help_cmd):
                self.assertTrue(self.cou(help_cmd).returncode == 0)
```

Reply:
I did not know about [SubTest](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests), that's really nice <3. 

> is a better approach because it will show which sub test failed.

If you use fixtures and parametrize from pytest it show also which tests and option failed and code is a little bit nicer with decorator instead of context manager. But I really like the subTest.

Reply:
I think my sentence was poor. I meant that was a better approach than what I had before and not that is a better approach than pytest or  parametrized that also has this feature of showing which sub test failed.